contracts/contracts/strategies/CurvePoolBooster.sol

```diff
@@ -0,0 +1,309 @@
+        );
+
+        emit TotalRewardAmountUpdated(balance);
+    }
+
+    /// @notice Manage the number of periods of the campaign
+    /// @dev This function should be called after the campaign is created
+    /// @param extraNumberOfPeriods Number of additional periods (cannot be 0)
+    /// @param bridgeFee Fee to pay for the bridge
+    /// @param additionalGasLimit Additional gas limit for the bridge
+    function manageNumberOfPeriods(
+        uint8 extraNumberOfPeriods,
+        uint256 bridgeFee,
+        uint256 additionalGasLimit
+    ) external onlyGovernorOrStrategist {
+        require(campaignId != 0, "Campaign not created");
+        require(extraNumberOfPeriods > 0, "Invalid number of periods");
+
+        // Manage the campaign
+        ICampaingRemoteManager(campaignRemoteManager).manageCampaign{
+            value: bridgeFee
+        }(
+            ICampaingRemoteManager.CampaignManagementParams({
+                campaignId: campaignId,
+                rewardToken: rewardToken,
+                numberOfPeriods: extraNumberOfPeriods,
+                totalRewardAmount: 0,
+                maxRewardPerVote: 0
+            }),
+            targetChainId,
+            additionalGasLimit
+        );
+
+        emit NumberOfPeriodsUpdated(extraNumberOfPeriods);
+    }
+
+    /// @notice Manage the reward per vote of the campaign
+    /// @dev This function should be called after the campaign is created
+    /// @param newMaxRewardPerVote New maximum reward per vote
+    /// @param bridgeFee Fee to pay for the bridge
+    /// @param additionalGasLimit Additional gas limit for the bridge
+    function manageRewardPerVote(
+        uint256 newMaxRewardPerVote,
+        uint256 bridgeFee,
+        uint256 additionalGasLimit
+    ) external onlyGovernorOrStrategist {
+        require(campaignId != 0, "Campaign not created");
+        require(newMaxRewardPerVote > 0, "Invalid reward per vote");
+
+        // Manage the campaign
+        ICampaingRemoteManager(campaignRemoteManager).manageCampaign{
+            value: bridgeFee
+        }(
+            ICampaingRemoteManager.CampaignManagementParams({
+                campaignId: campaignId,
+                rewardToken: rewardToken,
+                numberOfPeriods: 0,
+                totalRewardAmount: 0,
+                maxRewardPerVote: newMaxRewardPerVote
+            }),
+            targetChainId,
+            additionalGasLimit
+        );
+
+        emit RewardPerVoteUpdated(newMaxRewardPerVote);
+    }
+
+    /// @notice calculate the fee amount and transfer it to the feeCollector
+    /// @param amount Amount to calculate the fee
+    /// @return Amount after fee
+    function _handleFee(uint256 amount) internal returns (uint256) {
+        uint256 feeAmount = (amount * fee) / BASE_FEE;
+
+        // If there is a fee, transfer it to the feeCollector
+        if (feeAmount > 0) {
+            IERC20(rewardToken).transfer(feeCollector, feeAmount);
```

Comment:
nit: let's add a event for fee transfers

Reply:
https://github.com/OriginProtocol/origin-dollar/pull/2327/commits/8f1422470326d83057b3003093dbb155487253ee